pymeasure/instruments/advantest/advantestR624X.py

```diff
@@ -0,0 +1,2086 @@
+
+
+class AdvantestR624X(Instrument):
+    """ Represents the Advantest R624X series (channel A and B) SourceMeter and provides a
+    high-level interface for interacting with the instrument.
+
+    This is the base class for both AdvantestR6245 and AdvantestR6246 devices. It's not
+    necessary to instantiate this class directly instead create an instance of the
+    AdvantestR6245 or AdvantestR6246 class as shown in the following example:
+
+    .. code-block:: python
+
+        smu = AdvantestR6246("GPIB::1")
+        smu.reset()                                                     # Set default parameters
+        smu.ch_A.current_source(source_range = CurrentRange.FIXED_60mA,
+                                source_value = 0,                       # Source current at 0 A
+                                voltage_compliance = 10)                # Voltage compliance at 10 V
+        smu.ch_A.enable_source()                                        # Enables the source output
+        smu.ch_A.measure_voltage()
+        smu.ch_A.current_change_source = 5e-3                           # Change to 5mA
+        print(smu.read_measurement())                                   # Read and print the voltage
+        smu.ch_A.standby()                                              # Put channel A in standby
+
+    """
+
+    def __init__(self, adapter, name="R624X Source meter Base Class", **kwargs):
+        super().__init__(adapter, name, **kwargs)
+        self.sequence = []
+        self.store_to_sequence = False
+        self.sequence_line_count = 0
+
+    def write(self, command, **kwargs):
+        if self.store_to_sequence:
+            self.append_sequence_command(command)
+        else:
+            super().write(command, **kwargs)
+
+    def check_errors(self):
+        errors = {
+            100: "A fan stop was detected.",
+            101: "Since the overload detection of the {0} channel was activated, it was set to"
+                 "standby.",
+            102: "Since the overheat detection of the {0} channel worked, I made it a standby.",
+            200: "Received an undefined command.",
+            201: "There is an error in the data format.",
+            210: "Received data outside the set range.",
+            211: "A command was received that cannot be executed in the current settings.",
+            221: "Data output buffer overflowed.",
+        }
+
+        error = self.ask('err?')
+        unit = int(error[0:2])
+        err = int(error[2:5])
+        channel = f'{"B" if unit > 1 else "A"}'
+
+        if err in errors:
+            message = errors[err].format(channel)
+        elif err > 0 and err < 100:
+            if unit == 0:
+                message = "As a result of the self-test, an abnormality was found in the logic" \
+                          "part."
+            else:
+                message = "Result of self-test {0}-channel was found to be abnormal.".format(
```

Comment:
Just for your information,you do not have to change the code. The following work exactly as your current line:
```suggestion
                message = f"Result of self-test {channel}-channel was found to be abnormal."
```
The f in front of the quotes indicates to format the string. In the curly braces you can place any valid python expression, for example a variable name.

Oh I see, that you use it already below.

Reply:
I'll change that